src/charmed_kubeflow_chisme/testing/cos_integration.py

```diff
@@ -109,22 +102,30 @@ async def deploy_and_assert_grafana_agent(
     await model.wait_for_idle(apps=[GRAFANA_AGENT_APP], status="blocked", timeout=5 * 60)
 
 
-async def _check_metrics_endpoint(app: Application, metrics_endpoint: str) -> None:
-    """Check metrics endpoint accessibility.
+def _check_url(url: str, port: int, path: str) -> bool:
+    """Check port and path in url."""
+    output = urlparse(url)
+    return output.port == port and output.path == path
 
-    Checking accessibility of metrics endpoint from grafana-agent-k8s. If metrics endpoint is
-    defined as `*:5000/metrics` it will be changed to `<app-name>.<namespace>.svc:5000/metrics`.
+
+async def _check_metrics_endpoint(app: Application, port: int, path: str) -> None:
```

Comment:
Maybe rename it to make sure it reflects what we are doing.

```suggestion
async def _check_targets_endpoint(app: Application, port: int, path: str) -> None:
```

Or maybe even

```suggestion
async def _assert_targets_are_configured_in_endpoint(app: Application, port: int, path: str) -> None:
```